tools/create-local-npm-package.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-console */
+const { execSync } = require('child_process');
+const fs = require('fs');
+const { join } = require('path');
+
+const TAG = '[create-local-npm-package]';
+const PROJECT_ROOT = join(__dirname, '../'); // step out of the tools dir to project root
+const TMP_DIR = join(PROJECT_ROOT, './.tmp');
+
+const main = async () => {
+  if (!fs.existsSync(TMP_DIR)) {
+    fs.mkdirSync(TMP_DIR);
+  }
+
+  const nodeVersion = execSync('node --version')
+    .toString()
+    .trim();
+
+  const npmVersion = execSync('npm --version')
+    .toString()
+    .trim();
+
+  console.log(`${new Date().toJSON()} ${TAG} - NodeVersion=${nodeVersion}, npmVersion=${npmVersion}`);
+
+  const packageFileName = execSync('npm pack', { cwd: PROJECT_ROOT })
+    .toString()
+    .trim();
+
+  console.log(`${new Date().toJSON()} ${TAG} - packageFileName: "${packageFileName}"`);
+
+  const packageJsonRaw = fs.readFileSync(join(PROJECT_ROOT, 'package.json'));
+  const packageJson = JSON.parse(packageJsonRaw);
+  console.log(`${new Date().toJSON()} ${TAG} - packageJson.version=${packageJson.version}`);
+
+  const newPackageFileName = packageFileName.replace(packageJson.version, 'dev');
+  fs.renameSync(join(PROJECT_ROOT, packageFileName), join(TMP_DIR, newPackageFileName));
+  return newPackageFileName;
+};
+
+main()
+  .then(newPackageFileName => {
+    console.log(`${new Date().toJSON()} ${TAG} - SUCCESS_4d21b660-7676-4394-b853-642ce3617bed ${newPackageFileName}`);
+    process.exit(0);
+  })
+  .catch(ex => {
+    console.error(`${new Date().toJSON()} ${TAG} - Process crashed:`, ex);
```

Comment:
I would have to pull in dependencies for that, create a package.json, lock file, have people execute npm install in yet another sub-folder. 
It is by design that the only thing the script depends on is NodeJS being installed on the host OS so that it's short, self-contained, simple, etc.
http://sscce.org/

The latter actually goes for the `tools` directory as a whole as well. It is kept outside of the src/ folder so that it can contain 100% independent tools that don't depend on the software itself (that's what I'm aiming for at least, of course sometimes you cannot avoid certain things, but managed to get by so far).